From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Apthorp <jeremya@chromium.org>
Date: Mon, 15 Apr 2019 18:12:08 -0700
Subject: Export symbols needed for Windows build

These symbols are required to build v8 with BUILD_V8_SHARED on Windows.

diff --git a/src/objects/objects.h b/src/objects/objects.h
index 70efa0136023b47bcf054c3414170dff32be01dd..f967a2b62fe19d5f729bece878e0d93f2c4c0aea 100644
--- a/src/objects/objects.h
+++ b/src/objects/objects.h
@@ -823,7 +823,7 @@ enum class KeyCollectionMode {
 // Utility superclass for stack-allocated objects that must be updated
 // on gc.  It provides two ways for the gc to update instances, either
 // iterating or updating after gc.
-class Relocatable {
+class V8_EXPORT_PRIVATE Relocatable {
  public:
   explicit inline Relocatable(Isolate* isolate);
   inline virtual ~Relocatable();
diff --git a/src/objects/ordered-hash-table.h b/src/objects/ordered-hash-table.h
index 5f3c45a110aa48cf7c9c189d0e37e07193f06b75..01bfb2a73dbfb6d5a5d41baea946c783abfd1558 100644
--- a/src/objects/ordered-hash-table.h
+++ b/src/objects/ordered-hash-table.h
@@ -60,7 +60,7 @@ namespace internal {
 //                      of the removed holes.
 //   [kPrefixSize + 3 + NumberOfRemovedHoles()..length]: Not used
 template <class Derived, int entrysize>
-class OrderedHashTable : public FixedArray {
+class V8_EXPORT_PRIVATE OrderedHashTable : public FixedArray {
  public:
   // Returns an OrderedHashTable (possibly |table|) with enough space
   // to add at least one new element.
